File: src/main/java/com/EmployeesREST/jms/EmployeeJmsConsumer.java
package com.EmployeesREST.jms;

import com.EmployeesREST.config.JmsConfig;
import com.EmployeesREST.dao.EmployeeRepository;
import com.EmployeesREST.dto.Employee;
import lombok.extern.slf4j.Slf4j;
import org.springframework.jms.annotation.JmsListener;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class EmployeeJmsConsumer {
    private final EmployeeRepository employeeRepository;
    public EmployeeJmsConsumer(EmployeeRepository employeeRepository) {
        this.employeeRepository = employeeRepository;
    }

    @JmsListener(destination = JmsConfig.EMPLOYEES_QUEUE)
    public void consumeMessage(Employee employee) {
        log.info("IN: ConsumeMessage received employee {}", employee);
        employeeRepository.save(employee);
        log.info("OUT: Employee added: {}", employee);
    }

/*    @JmsListener(destination = JmsConfig.EMPLOYEES_QUEUE)
    public void consumeMessage(String employee) {
        log.info("Message received from activemq queue: " + employee);
        System.out.println(employee);
    }*/
}

File: src/main/kotlin/com/employees/jms/EmployeeJmsConsumer.kt
package com.employees.jms

import com.employees.config.JmsConfig
import com.employees.dao.EmployeeRepository
import com.employees.dto.Employee
import mu.KotlinLogging
import org.springframework.boot.autoconfigure.domain.EntityScan
import org.springframework.jms.annotation.JmsListener
import org.springframework.jms.core.JmsTemplate
import org.springframework.stereotype.Component

private val log = KotlinLogging.logger {}

@Component
@EntityScan
class EmployeeJmsConsumer(private val employeeRepository: EmployeeRepository, jmsTemplate: JmsTemplate) {
    @JmsListener(destination = JmsConfig.EMPLOYEES_QUEUE)
    fun consumeMessage(employee: Employee?) {
        log.info("IN: ConsumeMessage received employee {}", employee)
        employeeRepository.save(employee!!)
        log.info("OUT: Employee added: {}", employee)
    }
}